conftest.py

```diff
@@ -602,20 +609,17 @@ def pytest_collection_modifyitems(items, config):
             if deselect_test:
                 logger.info("SKIP: Deselecting non-upgrade test %s because of --execute-upgrade-tests-only" % item.name)
 
-        if item.get_closest_marker("resource_intensive") and not collect_only:
-            force_resource_intensive = config.getoption("--force-resource-intensive-tests")
-            skip_resource_intensive = config.getoption("--skip-resource-intensive-tests")
-            if not force_resource_intensive:
-                if skip_resource_intensive:
-                    deselect_test = True
-                    logger.info("SKIP: Deselecting test %s as test marked resource_intensive. To force execution of "
-                          "this test re-run with the --force-resource-intensive-tests command line argument" % item.name)
+        if item.get_closest_marker("resource_intensive"):
+            if skip_resource_intensive:
+                deselect_test = True
+                logger.info("SKIP: Deselecting test %s as test marked resource_intensive. To force execution of "
+                            "this test re-run with the --force-resource-intensive-tests command line argument" % item.name)
+            elif not force_resource_intensive:
                 if not sufficient_system_resources_resource_intensive:
                     deselect_test = True
                     logger.info("SKIP: Deselecting resource_intensive test %s due to insufficient system resources" % item.name)
 
-        if not item.get_closest_marker("resource_intensive") and not collect_only:
-            only_resource_intensive = config.getoption("--only-resource-intensive-tests")
+        if not item.get_closest_marker("resource_intensive"):
```

Comment:
I wonder what was the rationale for running this logic only if `not collect_only`?
maybe there was a valid reason not to do it always?
@michaelsembwever ?

Reply:
@tlasica It was already mentioned as a problem in https://issues.apache.org/jira/browse/CASSANDRA-16221

Reply:
Not that I can remember, unfortunately :(